src/ceph-disk/ceph_disk/main.py

```diff
@@ -1983,6 +1957,44 @@ def set_subparser(subparsers):
         )
         return parser
 
+    def prepare_metadata(self):
+        if self.args.osd_uuid is None:
+            self.args.osd_uuid = str(uuid.uuid4())
+
+        # prepare out secrets
+        keys = {
+            'cephx_secret', command('ceph-authtool', '--gen-print-key'),
+        };
+        if args.dmcrypt:
+            key_size = CryptHelpers.get_dmcrypt_keysize(self.args)
+            key = open('/dev/urandom', 'rb').read(key_size / 8)
+            keys['dmcrypt_key'] = base64.b64encode(key)
+            keys['cephx_lockbox_secret'] = command(
+                'ceph-authtool', '--gen-print-key')
+
+        # allocate an osd id from the mon
+        bootstrap = self.args.prepare_key_template.format(cluster=cluster,
+                                                          statedir=STATEDIR)
+        args = [
+            'ceph',
+            '--cluster', cluster,
+            '--name', 'client.bootstrap-osd',
+            '--keyring', bootstrap,
+            'osd',
+            'new',
+            args.osd_uuid,
+        ]
+        if args.osd_replace_id:
```

Comment:
I think that `--replace-osd-id` becomes `args.replace_osd_id` unless you specify you really want it in `args.osd_replace_id`